src/averageAges.js

```diff
@@ -55,7 +83,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const getPerson = onlyWithSon
+    ? people
+      .filter(person => person.sex === 'm')
+    : people;
+
+  let len = 0;
+
+  const sum = getPerson.reduce((obj, person) => {
+    const mother = people.find(x => x.name === person.mother);
+
+    if (mother) {
+      len++;
+
+      return obj + person.born - mother.born;
+    }
+
+    return obj;
+  }, 0);
+
+  return sum / len;
```

Comment:
Again, it's very hard to read. Create variables, move some logic to separate functions etc.